File: examples/python/basic.py
"""
Variable Name Converter API - Basic Usage Example

This example demonstrates the basic usage of the Variable Name Converter API.
API Documentation: https://docs.apiverve.com/ref/variablenameconverter
"""

import os
import requests
import json

API_KEY = os.getenv('APIVERVE_API_KEY', 'YOUR_API_KEY_HERE')
API_URL = 'https://api.apiverve.com/v1/variablenameconverter'

def call_variablenameconverter_api():
    """
    Make a POST request to the Variable Name Converter API
    """
    try:
        # Request body
        request_body &#x3D; {
    &#x27;variables&#x27;: [
        &#x27;myVariableName&#x27;,
        &#x27;another_var_name&#x27;,
        &#x27;SomeClassName&#x27;,
        &#x27;kebab-case-name&#x27;,
        &#x27;CONSTANT_VALUE&#x27;
    ],
    &#x27;to&#x27;: &#x27;snake_case&#x27;
}

        headers = {
            'x-api-key': API_KEY,
            'Content-Type': 'application/json'
        }

        response = requests.post(API_URL, headers=headers, json=request_body)

        # Raise exception for HTTP errors
        response.raise_for_status()

        data = response.json()

        # Check API response status
        if data.get('status') == 'ok':
            print('✓ Success!')
            print('Response data:', json.dumps(data['data'], indent=2))
            return data['data']
        else:
            print('✗ API Error:', data.get('error', 'Unknown error'))
            return None

    except requests.exceptions.RequestException as e:
        print(f'✗ Request failed: {e}')
        return None

if __name__ == '__main__':
    print('📤 Calling Variable Name Converter API...\n')

    result = call_variablenameconverter_api()

    if result:
        print('\n📊 Final Result:')
        print(json.dumps(result, indent=2))
    else:
        print('\n✗ API call failed')

File: examples/javascript/basic.js
/**
 * Variable Name Converter API - Basic Usage Example
 *
 * This example demonstrates the basic usage of the Variable Name Converter API.
 * API Documentation: https://docs.apiverve.com/ref/variablenameconverter
 */

const API_KEY = process.env.APIVERVE_API_KEY || 'YOUR_API_KEY_HERE';
const API_URL = 'https://api.apiverve.com/v1/variablenameconverter';

/**
 * Make a POST request to the Variable Name Converter API
 */
async function callVariableNameConverterAPI() {
  try {
    // Request body
    const requestBody &#x3D; {
    &quot;variables&quot;: [
        &quot;myVariableName&quot;,
        &quot;another_var_name&quot;,
        &quot;SomeClassName&quot;,
        &quot;kebab-case-name&quot;,
        &quot;CONSTANT_VALUE&quot;
    ],
    &quot;to&quot;: &quot;snake_case&quot;
};

    const response = await fetch(API_URL, {
      method: 'POST',
      headers: {
        'x-api-key': API_KEY,
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(requestBody)
    });

    // Check if response is successful
    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();

    // Check API response status
    if (data.status === 'ok') {
      console.log('✓ Success!');
      console.log('Response data:', data.data);
      return data.data;
    } else {
      console.error('✗ API Error:', data.error || 'Unknown error');
      return null;
    }

  } catch (error) {
    console.error('✗ Request failed:', error.message);
    return null;
  }
}

// Run the example
callVariableNameConverterAPI()
  .then(result => {
    if (result) {
      console.log('\n📊 Final Result:');
      console.log(JSON.stringify(result, null, 2));
    }
  });
